aws/resource_aws_ebs_snapshot_import_test.go

```diff
@@ -0,0 +1,216 @@
+			if response.Snapshots != nil && len(response.Snapshots) > 0 {
+				*v = *response.Snapshots[0]
+				return nil
+			}
+		}
+		return fmt.Errorf("Error finding EC2 Snapshot %s", rs.Primary.ID)
+	}
+}
+
+func testAccCheckAWSEbsSnapshotImportDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ebs_snapshot_import" {
+			continue
+		}
+		input := &ec2.DescribeSnapshotsInput{
+			SnapshotIds: []*string{aws.String(rs.Primary.ID)},
+		}
+
+		output, err := conn.DescribeSnapshots(input)
+		if err != nil {
+			if isAWSErr(err, "InvalidSnapshot.NotFound", "") {
+				continue
+			}
+			return err
+		}
+		if output != nil && len(output.Snapshots) > 0 && aws.StringValue(output.Snapshots[0].SnapshotId) == rs.Primary.ID {
+			return fmt.Errorf("EBS Snapshot %q still exists", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccAwsEbsSnapshotImportConfigBasic(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket" "images" {
+  bucket_prefix = "images-"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_object" "image" {
+  bucket = aws_s3_bucket.images.id
+  key = "diskimage.vhd"
+  content_base64 = %[1]q
+}
+
+# The following resources are for the *vmimport service user*
+# See: https://docs.aws.amazon.com/vm-import/latest/userguide/vmie_prereqs.html#vmimport-role
+resource "aws_iam_role" "vmimport" {
+  assume_role_policy = data.aws_iam_policy_document.vmimport-trust.json
+}
+
+resource "aws_iam_role_policy" "vmimport-access" {
+  role = aws_iam_role.vmimport.id
+  policy = data.aws_iam_policy_document.vmimport-access.json
+}
+
+data "aws_iam_policy_document" "vmimport-access" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetBucketLocation",
+      "s3:GetObject",
+      "s3:ListBucket",
+    ]
+    resources = [
+      aws_s3_bucket.images.arn,
+      "${aws_s3_bucket.images.arn}/*"
+    ]
+  }
+  statement {
+    effect = "Allow"
+    actions = [
+      "ec2:ModifySnapshotAttribute",
+      "ec2:CopySnapshot",
+      "ec2:RegisterImage",
+      "ec2:Describe*"
+    ]
+    resources = [
+      "*"
+    ]
+  }
+}
+
+
+data "aws_iam_policy_document" "vmimport-trust" {
+  statement {
+    effect = "Allow"
+    principals {
+      type = "Service"
+      identifiers = [ "vmie.amazonaws.com" ]
+    }
+
+    actions = [
+      "sts:AssumeRole"
+    ]
+
+    condition {
+      test = "StringEquals"
+      variable = "sts:ExternalId"
+      values = [ "vmimport" ]
+    }
+  }
+}
+
+resource "aws_ebs_snapshot_import" "test" {
+  disk_container {
+    description = %[2]q
+    format = "VHD"
+    user_bucket {
+      s3_bucket = aws_s3_bucket.images.id
+      s3_key = aws_s3_bucket_object.image.key
+    }
+  }
+
+  role_name = aws_iam_role.vmimport.name
+
+  timeouts {
+    create = "10m"
+    delete = "10m"
+  }
+}
+`, testAccAwsEbsSnapshotDisk(), rName)
+}
+
+func testAccAwsEbsSnapshotDisk() string {
```

Comment:
This is a bit funny probably, but to test importing a disk image we need a disk image to import. The image on disk, base64'd, is about 3400 bytes long. Compressing, then base64'ing is much shorter, with the trade-off that we need to decompress and rebase64 on the fly for the test.